crates/uv-pypi-types/Cargo.toml

```diff
@@ -29,10 +29,12 @@ indexmap = { workspace = true, features = ["serde"] }
 itertools = { workspace = true }
 jiff = { workspace = true, features = ["serde"] }
 mailparse = { workspace = true }
+petgraph = { workspace = true }
 regex = { workspace = true }
 rkyv = { workspace = true }
+rustc-hash = { workspace = true }
 schemars = { workspace = true, optional = true }
-serde = { workspace = true }
+serde = { workspace = true, optional = true }
```

Comment:
How come this was made optional? I'm kinda wondering how this works at all, since I don't see any features enabling `serde` in this crate? I guess feature unification or something?

Reply:
It's because of moving `DependencyGroups` into this crate in order to depend on it in `conflicts.rs`. It looks like my comment on that file disappeared from this PR.

Reply:
Does it have to be optional? (I'm overall kinda skeptical of even having optional dependencies in our crates unless there is a very specific need for them. In practice, in most cases, all of the features end up enabled through feature unification.)

Reply:
(In general I think it's fine -- it's a pattern we use for Serde and Schemars, so I have no objection to propagating it. We should revisit elsewhere if we want to move away from it.)

Reply:
I am slightly confused about why it was not optional before, but is optional now. Maybe that's Andrew's question.

Reply:
The superficial reason is because the compiler complained when I moved over `dependency-groups.rs`. But I think it's because of the fact that `DependencyGroups` is used a field on a struct with a feature flag in `pyproject.rs`.